walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,82 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
-public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+
+public final class FindMeetingQuery {   
+
+    /* Returns sorted TimeRanges that have attendees in the attendees set. */
+    private static ArrayList<TimeRange> getBusyTimeRangesForAttendees(Collection<Event> events, Set<String> attendeesSet) { 
+        ArrayList<TimeRange> timeRangesWithAttendees = new ArrayList <TimeRange> ();
+        for(Event event: events) {
+            Collection<String> eventAttendees = event.getAttendees();
+            TimeRange when = event.getWhen();
+            for(String attendee : eventAttendees) {   
+                if(attendeesSet.contains(attendee)) {
+                    timeRangesWithAttendees.add(when);
+                    break;
+                } 
+            }
+        }
+        Collections.sort(timeRangesWithAttendees, TimeRange.ORDER_BY_START);
+        return timeRangesWithAttendees;
+    }
+
+    //Calculates the availbleTimes from the series of events
+    private static ArrayList <TimeRange> findAvailableTimes(Collection<Event> events, Collection<String> mandatoryAttendees, long duration) {
+      ArrayList<TimeRange> availableTimeForAll = new ArrayList<TimeRange>();
+      int end = TimeRange.END_OF_DAY;
```

Comment:
this variable is only used once on line 85. Consider moving this closer to where it is used (makes the code easier to read) or to use TimeRange.END_OF_DAY directly.